php-classes/Emergence/Connectors/Job.php

```diff
@@ -160,40 +146,71 @@ public function logRecordDelta(ActiveRecord $Record, $options = array())
 
         $logEntry['message'] = call_user_func($messageRenderer, $logEntry);
 
-        return $this->log[] = $logEntry;
+        return $this->log(
+            $logEntry['level'],
+            $logEntry['message'],
+            [
+                'changes' => $changes,
+                'record' => $Record
+            ]
+        );
     }
 
     public function logInvalidRecord(\ActiveRecord $Record)
     {
-        return $this->log(array(
-            'message' => 'Invalid '.get_class($Record).' record: '.$Record->getTitle()
-            ,'validationErrors' => $Record->validationErrors
-        ), LogLevel::WARNING);
+        return $this->log(
+            LogLevel::WARNING,
+            'Invalid '.get_class($Record).' record: '.$Record->getTitle(),
+            [
+                'validationErrors' => $Record->validationErrors
+            ]
+        );
     }
 
     public function logException(\Exception $e)
     {
-        return $this->log(array(
-            'message' => get_class($e).': '.$e->getMessage()
-            ,'exception' => $e
-        ), LogLevel::ERROR);
+        return $this->log(
+            LogLevel::ERROR,
+            'Exception:: '.get_class($e).': '.$e->getMessage(),
```

Comment:
@nbey let's parameterize this message `'Exception({exceptionClass}): {exceptionMessage}'`